### PR TITLE
KAS-4608: Automatically order agendaitems

### PR DESCRIPTION
**Pull request by sergiofenoll**
https://kanselarij.atlassian.net/browse/KAS-4608

When submitting a subcase on an agenda, we now list all the agendaitems related to said specific agenda version (i.e. the latest ontwerp agenda of a meeting) and order them by mandatee priority.

Items without a mandatee (e.g. decreten) are sorted at the end of the list.

When a new agenda version is created, new items will not take into account the order of items of previous versions. I.e. new mandatee priority groups will be "made".

### CODE REVIEW

File: lib/agendaitem-order.js [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+  const sortedAgendaitems = agendaitems.sort(
+    (a1, a2) => {
+      const priority1 = Array.isArray(a1.mandateePriority) ? a1.mandateePriority.sort().toString() : String(a1.mandateePriority);
+      const priority2 = Array.isArray(a2.mandateePriority) ? a2.mandateePriority.sort().toString() : String(a2.mandateePriority);
+      return priority1 === priority2
+        ? a1.agendaitemPosition - a2.agendaitemPosition
+        : priority1 < priority2
+          ? -1
+          : 1
+    });
```
Comment (ValenberghsSven) [inline lib/agendaitem-order.js]: does this sorting also move the approval item?
If present, that should stay as the first agendaitem.

Comment (ValenberghsSven) [inline lib/agendaitem-order.js]: Does the sort on numbering work with 11 mandatees?
f.e. the group [1, 2]  is more important than the group [1, 10, 11]
That is an issue we encountered in newsletter service and to solve that we used letters instead of numbers.

Comment (sergiofenoll) [reply lib/agendaitem-order.js]: The query to fetch the agendaitems of the latest agenda has the following bit: 
```sparql
  ?agendaActivity besluitvorming:genereertAgendapunt ?agendaitem ;
    besluitvorming:vindtPlaatsTijdens ?subcase .
```
https://github.com/kanselarij-vlaanderen/agenda-submission-service/pull/4/files#diff-5b6ddf4aacad6a5d676d764ced5b8b1a51d3ce8a3df4d7523b8fe7c450a6c983R21-R22

The approval shouldn't ever be found because it has no agenda activity.

Comment (sergiofenoll) [reply lib/agendaitem-order.js]: Ah, didn't account for that. Will use the letter based approach.

Comment (sergiofenoll) [reply lib/agendaitem-order.js]: Dealt with here https://github.com/kanselarij-vlaanderen/agenda-submission-service/pull/4/commits/9a67f98e01bc2e7879a28baa51b66e20e88ab507 and here https://github.com/kanselarij-vlaanderen/agenda-submission-service/pull/4/commits/e22fa31f07015d133c2c6dfcdc8ca96afb5b8f42

Not sure how often (if at all?) there will be more than 9 ministers, but I went for a more "general" approach. Instead of replacing numbers with letters (which would still limit us to 26 ministers), the priority number is converted to a string which is then padded with zeroes according to the longest number that was encountered.
So `[[1, 2], [1, 10, 11]]` will get sorted as `[['01', '02'], ['01', '10', '11']]`

Comment (ValenberghsSven) [reply lib/agendaitem-order.js]: From what I remember, there can only be max 11 ministers.
Direct quote from VP website confirms this:
"De Vlaamse Regering bestaat uit maximaal elf ministers. De huidige Vlaamse Regering telt negen ministers"

Comment (ValenberghsSven) [reply lib/agendaitem-order.js]: ah yes. this paired with the `math.min` should work.
The only time this would not work is when the approval item is not the first agendaitem.
It shouldn't really happen afaik but there are 2 old agendas in PROD where this is the case.
We could ask the question if this will ever happen again.
But if it does, the approval item will have a duplicate position after this code since we don't take it into account?

**Review by tdn — APPROVED** [review]
Works well locally, no errors.
More extensive tests for edge-cases can be done on DEV or ACCEPT.

File: lib/agendaitem-order.js (hunk 1 above)
Comment (tdn) [reply lib/agendaitem-order.js]: Yes, just tested, and if the approval is moved to position 2, and then another agendaitem is added, you get 2 agendaitems with position 2.

Now, since moving the approval is clearly a user error, and it is very easily fixed by simply moving it back in place (or even to some other place on the agenda), I'm inclined to merge this as such, unless you really insist to have this fixed @ValenberghsSven ?

**Comment by tdn** [comment]
Added a few things. The sorting within one mandatee "block" was random, due to a missing toString. 

Also added a filter for the last approved agendaitem, to start the numbering from there.

**Comment by tdn** [comment]
Currently we also sort the announcements. This merge is pending on whether this is desired by the users.

**Review by ValenberghsSven — APPROVED** [review]
Tested it with Tom, found some scenarios that didn't work and fixed them.
Should be all good to go
